jerry-core/ecma/operations/ecma-dataview-object.c

```diff
@@ -145,13 +131,22 @@ ecma_op_dataview_create (const ecma_value_t *arguments_list_p, /**< arguments li
                                                 sizeof (ecma_dataview_object_t),
                                                 ECMA_OBJECT_TYPE_CLASS);
 
+  ecma_deref_object (prototype_obj_p);
+
+  /* 10. */
+  if (ecma_arraybuffer_is_detached (buffer_p))
+  {
+    ecma_deref_object (object_p);
```

Comment:
Wait for a second. This leads to undefined behavior in GC. Since the allocated object's type is `ECMA_OBJECT_TYPE_CLASS` the mark & sweep depends on it's internal `class_id` which is uninitialized at this point. So perform the deatched arraybuffer check **after** then the dataview object has been fully initialized.

Reply:
Good catch. We found a better solution, moving ecma_create_object call after 10.